sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1774,6 +1775,51 @@ class Analyzer(
   }
 
   /**
+   * Substitute Hints.
+   * - BROADCAST/BROADCASTJOIN/MAPJOIN match the closest table with the given name parameters.
+   */
+  object SubstituteHints extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan transform {
+      case logical: LogicalPlan => logical transformDown {
+        case h @ Hint(name, parameters, child)
+            if Seq("BROADCAST", "BROADCASTJOIN", "MAPJOIN").contains(name.toUpperCase) =>
+          var resolvedChild = child
+
+          for (param <- parameters) {
+            val names = param.split("\\.")
```

Comment:
See the PR: https://github.com/apache/spark/pull/14244 

I think we should ensure that all the cases in `TableIdentifierParserSuite.scala` if you do not want to parse it in `Analyzer`